Add parameters modification UI

CloeD opened the pull request:
**What's in the PR ?**
- Retrieve parameters from node at launch
   - max_attempts
   - timeout
   - time_dilation_factor
   - voxel_size
   - collision_activation_distance
- Parameters are set in real-time:
   - max_attempts
   - timeout
   - time_dilation_factor
- Parameters are set after pressing the Confirm Button
   - voxel_size
   - collision_activation_distance

-> Since the implementation is made with `SyncParametersClient`, user must wait after curobo_gen_traj initialization before launching Rviz.

When parameters are set after pressing the Confirm, curobo_gen_traj reloads in a few seconds (and not the 15 seconds of usual warmup).

**What's next ?**
- Disable the button when setting world parameters (voxel_size, collision_activation_distance)
    -  Requires threading

CloeD committed:
retrieve parameters from node at launch

CloeD committed:
add parameter set + trigger call

CloeD committed:
add voxel + collision parameters + cleanup

saxtot commented on src/curobo_rviz.cpp:
```diff
@@ -17,14 +20,17 @@ namespace curobo_rviz
+    }
 
     // Connect SpinBox and DoubleSpinBox to slots
     connect(ui_->spinBox_1, SIGNAL(valueChanged(int)), this, SLOT(updateMaxAttempts(int)));
     connect(ui_->doubleSpinBox_1, SIGNAL(valueChanged(double)), this, SLOT(updateTimeout(double)));
     connect(ui_->doubleSpinBox_2, SIGNAL(valueChanged(double)), this, SLOT(updateTimeDilationFactor(double)));
+    connect(ui_->doubleSpinBox_3, SIGNAL(valueChanged(double)), this, SLOT(updateVoxelSize(double)));
+    connect(ui_->doubleSpinBox_4, SIGNAL(valueChanged(double)), this, SLOT(updateCollisionActivationDistance(double)));
```
Could the UI elements have more distinguishable names instead?

will-44 replied:
+1

saxtot commented on src/curobo_rviz.cpp:
```diff
@@ -58,26 +56,87 @@ namespace curobo_rviz
+                                       rclcpp::Parameter("collision_activation_distance", collision_activation_distance_)});
+        RCLCPP_INFO(node_->get_logger(), "Parameters set:\voxel_size: %.2f, collision_activation_distance: %.2f", voxel_size_, collision_activation_distance_);
+        
+        // call Trigger service update_motion_gen_config
+        rclcpp::Client<std_srvs::srv::Trigger>::SharedPtr client = node_->create_client<std_srvs::srv::Trigger>("/curobo_gen_traj/update_motion_gen_config");
+        auto request = std::make_shared<std_srvs::srv::Trigger::Request>();
```
This client could be created a single time in the constructor instead of every time it's called. The variable could use a rename too.

saxtot commented on src/curobo_rviz.cpp:
```diff
@@ -58,26 +56,87 @@ namespace curobo_rviz
+        auto result = client->async_send_request(request);
+        RCLCPP_INFO(node_->get_logger(), "Service call sent.");
+        
+        if (rclcpp::spin_until_future_complete(node_, result) == rclcpp::FutureReturnCode::SUCCESS) {
+            RCLCPP_INFO(node_->get_logger(), "Service call successful.");
+        } else {
+            RCLCPP_ERROR(node_->get_logger(), "Service call failed.");
+        }
```
The [std_srvs/Trigger](http://docs.ros.org/en/noetic/api/std_srvs/html/srv/Trigger.html)'s response contains the `success` and `message` arguments. That information could be used in the logging for more accuracy. For example, the `success` argument could be `false`, but the `rclcpp::FutureReturnCode::SUCCESS` argument will be `true`, since that only checks that the service responded, not that it didn't get an exception in its execution.

will-44 commented on src/curobo_rviz.cpp:
```diff
@@ -58,26 +56,87 @@ namespace curobo_rviz
+        auto result = client->async_send_request(request);
+        RCLCPP_INFO(node_->get_logger(), "Service call sent.");
+        
+        if (rclcpp::spin_until_future_complete(node_, result) == rclcpp::FutureReturnCode::SUCCESS) {
```
Ici on regarde uniquement que le service a bien répondu à pas sa réponse, c'est bien ça ? Donc si la réponse est "failed", on considère que c'est réussi ?

will-44 reviewed (commented):
deux petites choses a modifier

CloeD committed:
refactor variable names for ui + update ui

CloeD committed:
create client in constructor + rename var

CloeD committed:
check if trigger successful + retrieve its message

CloeD committed:
add confirmation message after confirm changes is clicked

CloeD commented:
![image](https://github.com/user-attachments/assets/ecaf9884-ecf7-4943-8f16-5a8cfe5cdc49)
- Added labels to the UI for more clarity
- Added a confirmation message under the Confirm Changes button that is shown for 5 seconds (the message is taken from the Trigger service message)
